main/inc/lib/userportal.lib.php

```diff
@@ -1007,6 +1007,11 @@ public function return_navigation_links()
         // Tabs that are deactivated are added here.
         if (!empty($this->tpl->menu_navigation)) {
             foreach ($this->tpl->menu_navigation as $section => $navigation_info) {
+                
+                if(is_null($navigation_info)) {
```

Comment:
Nous utilisons des conventions de style dans Chamilo (partout). Ici, il manque un espace entre le terme conditionnel (if) et la parenthèse qui suit. Les espaces permettent de faire la différence entre des fonctions (`function if($hop) { ... }`) et des condionnelles (`if ($hop) { ... }`).
Merci de lire au moins une fois nos conventions ici https://github.com/chamilo/chamilo-lms/wiki/Coding-conventions, qui se régissent par défaut sur base de https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
Dans ce cas précis, on parle de PSR-2, section 5 "There MUST be one space after the control structure keyword"

Reply:
Aussi, nous utilisons peu `is_null()`, qui est un peu trop restrictif, et préférons `if (empty($navigation_info))` qui embarque plus de possibilités.
La question ici c'est surtout: est-ce qu'on veut vraiment un élément de navigation vide, et est-ce qu'on n'ignorerait pas plutôt tout simplement l'élément en faisant
```
if (empty($navigation_info)) {
    continue;
}
```
...qui aura pour effet d'ignorer cet élément en particulier et de passer au suivant. Sinon, en le remplissant de strings vides, on crée quand même un élément dans la navigation (donc les séparateurs s'afficheront), malgré qu'il soit vide...